DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -419,11 +419,60 @@ function Get-ChocoInstalledPackage ($action) {
         if ( $PackageCacheSec -lt (Get-Item $ChocoInstallList -ErrorAction SilentlyContinue).LastWriteTime ) {
                 $res = Import-Clixml $ChocoInstallList
         } else {
-            choco list -lo -r | ConvertFrom-Csv -Header 'Name', 'Version' -Delimiter "|" -OutVariable res | Export-Clixml -Path $ChocoInstallList
+            Invoke-Chocolatey "list -lo -r" | ConvertFrom-Csv -Header 'Name', 'Version' -Delimiter "|" -OutVariable res | Export-Clixml -Path $ChocoInstallList
         }
     }
 
     Return $res
 }
 
+<#
+.Synopsis
+    Invoke the chocolatey executable on the sytem. Throws choco error
+.DESCRIPTION
+    Invoke the chocolatey executable on the sytem. Throws choco error, also when used in DSC configurations.
+.EXAMPLE
+    Invoke-Chocolatey "list"
+.EXAMPLE
+    Invoke-Chocolatey "list -lo"
+.EXAMPLE
+    IChoco "list -lo"
+#>
+function Invoke-Chocolatey
+{
+    [CmdletBinding()]
+    [Alias("IChoco")]
+    [OutputType([int])]
+    Param
+    (
+        # Param1 help description
+        [Parameter(Position=0)]
+        $pParams
+    )
+    Process
+    {
+        Write-Verbose -Message "command: 'choco $pParams'"
+        $pinfo = New-Object System.Diagnostics.ProcessStartInfo
+        $pinfo.FileName = "choco"
+        $pinfo.RedirectStandardError = $true
+        $pinfo.RedirectStandardOutput = $true
+        $pinfo.UseShellExecute = $false
+        $pinfo.Arguments = $pParams
+        $p = New-Object System.Diagnostics.Process
+        $p.StartInfo = $pinfo
+        $p.Start() | Out-Null
+        $output = $p.StandardOutput.ReadToEnd()
+        $p.WaitForExit()
+
+        if($p.ExitCode -eq 0)
+        {
+            $output
+        }
+        else
+        {
+            throw $output
+        }
+    }
+}
+

```

Comment:
This probably can't influence the exit code of the run can it?

Reply:
exit code is now handled in pull request 103